### PR TITLE
Use is-in-mainnet in pox-4.clar

### PR DESCRIPTION
**Pull request by friedger**
### Description
This PR changes the way how pox-4 contracts for mainnet and testnet are created. Instead of string concatenation, we now use `is-in-mainnet` in clarity.

### Applicable issues

- addresses issue expressed in https://github.com/stacks-network/stacks-core/pull/4094#discussion_r1407961951 

### Additional info (benefits, drawbacks, caveats)
pox_4_body is only used to create pox-4 contract for mainnet and testnet. It is not used for creating pox-4 contract with other constants.

### Checklist

- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: contrib/core-contract-tests/Clarinet.toml [hunk 1]
```diff
@@ -9,11 +9,8 @@ costs_version = 1
 path = "../../stackslib/src/chainstate/stacks/boot/bns.clar"
 depends_on = []
 
-# pox-4 contains concat of boot/pox-4 and boot/pox-testnet
-# path = "../../stackslib/src/chainstate/stacks/boot/pox-4.clar"
-# path = "../../stackslib/src/chainstate/stacks/boot/pox-testnet.clar"
 [contracts.pox-4]
-path = "./contracts/pox/pox-4-testnet.clar"
+path = "../../stackslib/src/chainstate/stacks/boot/pox-4.clar"
```
Comment (moodmosaic) [inline contrib/core-contract-tests/Clarinet.toml:13]: 🎉

**Review by moodmosaic — COMMENTED** [review]
LGTM!

**Review by hugoclrd — APPROVED** [review]
LGTM! thx @friedger

**Review by jcnelson — COMMENTED** [review]
I wish I'd had a chance to review this before it got merged, because this patch could use some improvement.  Namely, the organization of the `ADDRESS_VERSION` constants are confusing.  If you look in `pox-4.clar`, you'll see code comments to the effect that the four address versions here are included in a separate file, and yet here they are.  Could you send another PR that cleans up `pox-4.clar`'s `ADDRESS_VERSION`s?

**Pull request merged**